ledger/participant-integration-api/src/test/lib/scala/platform/store/backend/StorageBackendTestsIngestion.scala

```diff
@@ -143,44 +143,49 @@ private[backend] trait StorageBackendTestsIngestion
   ) { connections =>
     import scala.concurrent.ExecutionContext.Implicits.global
 
-    val List(connection1, connection2) = connections
-    def packageFor(n: Int): DbDto.Package =
-      dtoPackage(offset(n.toLong))
-        .copy(
-          package_id = s"abc123$n",
-          _package = Random.nextString(Random.nextInt(200000) + 200).getBytes,
-        )
-    val conflictingPackageDtos = 11 to 20 map packageFor
-    val packages1 = 21 to 30 map packageFor
-    val packages2 = 31 to 40 map packageFor
-
-    def test() = {
-
-      executeSql(backend.parameter.initializeParameters(someIdentityParams))
-
-      def ingestPackagesF(connection: Connection, packages: Iterable[DbDto.Package]): Future[Unit] =
-        Future {
-          connection.setAutoCommit(false)
-          ingest(packages.toVector, connection)
-          connection.commit()
-        }
-
-      val ingestF1 = ingestPackagesF(connection1, packages1 ++ conflictingPackageDtos)
-      val ingestF2 = ingestPackagesF(connection2, packages2 ++ conflictingPackageDtos)
-
-      Await.result(ingestF1, Duration(10, "seconds"))
-      Await.result(ingestF2, Duration(10, "seconds"))
-
-      executeSql(updateLedgerEnd(offset(50), 0))
-
-      executeSql(backend.packageBackend.lfPackages).keySet.map(_.toString) shouldBe (
-        conflictingPackageDtos ++ packages1 ++ packages2
-      ).map(_.package_id).toSet
-    }
-
-    1 to NumberOfUpsertPackagesTests foreach { _ =>
-      test()
-      executeSql(backend.reset.resetAll)
+    inside(connections) { case List(connection1, connection2) =>
+      def packageFor(n: Int): DbDto.Package =
+        dtoPackage(offset(n.toLong))
+          .copy(
+            package_id = s"abc123$n",
+            _package = Random.nextString(Random.nextInt(200000) + 200).getBytes,
+          )
+
+      val conflictingPackageDtos = 11 to 20 map packageFor
+      val packages1 = 21 to 30 map packageFor
+      val packages2 = 31 to 40 map packageFor
+
+      def test() = {
+
+        executeSql(backend.parameter.initializeParameters(someIdentityParams))
+
+        def ingestPackagesF(
+            connection: Connection,
+            packages: Iterable[DbDto.Package],
+        ): Future[Unit] =
+          Future {
+            connection.setAutoCommit(false)
+            ingest(packages.toVector, connection)
+            connection.commit()
+          }
+
+        val ingestF1 = ingestPackagesF(connection1, packages1 ++ conflictingPackageDtos)
+        val ingestF2 = ingestPackagesF(connection2, packages2 ++ conflictingPackageDtos)
+
+        Await.result(ingestF1, Duration(10, "seconds"))
```

Comment:
not the great way to deal with futures in scalatest, there is trait `ScalaFutures` which allows to call `.futureValue` on Future and it supports timeout scaling through patience configuration and is also recommended approach.

I see though your change is not related to that await, im fine if you ignore me :)